src/sdk/Player.js

```diff
@@ -184,24 +184,39 @@ export class Player {
           this.destinationLocation = { x: winner.x, y: winner.y };
         }
 
-      }else {
+      } else {
         this.hasLOS = LineOfSight.hasLineOfSightOfMob(region, this.location.x, this.location.y, this.seeking, this.attackRange());
-        if (!this.hasLOS){
+        if (!this.hasLOS) {
           const seekingTiles = [];
-          for (let xx=0; xx < this.seeking.size; xx++){
-            for (let yy=0; yy < this.seeking.size; yy++){
-              seekingTiles.push({
-                x: this.seeking.location.x + xx, 
-                y: this.seeking.location.y - yy
-              });
+          // "When clicking on an npc, object, or player, the requested tiles will be all tiles"
+          // "within melee range of the npc, object, or player."
+          for (let xx=-1; xx <= this.seeking.size; xx++){
+            for (let yy=-1; yy <= this.seeking.size; yy++){
+              // Edges only, and no corners.
+              if ((xx == -1 || xx == this.seeking.size || yy == -1 || yy == this.seeking.size)
+                && ((xx != yy) && (xx != -1 || yy != this.seeking.size) && (xx != this.seeking.size || yy != -1))) {
+                // Don't path into an unpathable object.
+                const px = this.seeking.location.x + xx;
+                const py = this.seeking.location.y - yy;
+                if (!Pathing.collidesWithAnyEntities(region, px, py, 1)) {
+                  seekingTiles.push({
+                    x: px,
+                    y: py
+                  });
+                }
+              }
             }
           }
           // Create paths to all npc tiles
-          const potentialPaths = _.map(seekingTiles, (point) => Pathing.constructPath(region, this.location, { x: point.x, y: point.y }).length);
+          const potentialPaths = _.map(seekingTiles, (point) => Pathing.constructPath(region, this.location, { x: point.x, y: point.y }));
+          const validPaths = _.filter(potentialPaths, (path) => {
```

Comment:
I believe this filter isn't doing anything

Reply:
True, it _was_ doing something before, but I pulled it out